### PR TITLE
[bitnami/redis] Add Topology Spread Constraints

### PR DESCRIPTION
**Pull request by jonstelly**
**Description of the change**

This adds support for specifying [topology spread constraints](https://kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/) to ensure that redis pods are reasonably distributed across a cluster.

**Benefits**

In the simplest case, can be used to encourage better pod distribution across cluster nodes.  The `topologyKey` can be any node label, so for cloud providers you can spread across [region or zone](https://kubernetes.io/docs/reference/kubernetes-api/labels-annotations-taints/#topologykubernetesiozone), etc...

**Possible drawbacks**

Topology spread applies to each stateful set independently.  Since the redis master is one stateful set and the slave is another, it's still possible that a slave ends up on the same node as the master.  That could be fixed by combining with some other affinity/taints if needed.

**Applicable issues**

  - fixes #

**Additional information**
This is a resubmitted version of https://github.com/bitnami/charts/pull/2521 - I couldn't "Allow edits by maintainers" since I had forked into an organization.

***Examples***

Node Spread

```yaml
slave:
  spreadConstraints:
    - maxSkew: 1
      topologyKey: node
      whenUnsatisfiable: DoNotSchedule
```

**Checklist** 
- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[bitnami/chart]`)
- [x] If the chart contains a `values-production.yaml` apart from `values.yaml`, ensure that you implement the changes in both files

:warning: Keep in mind that if you want to make changes to the kubeapps chart, please implement them in the [kubeapps repository](https://github.com/kubeapps/kubeapps/tree/master/chart/kubeapps). This is only a synchronized mirror.

### CODE REVIEW

**Comment by jonstelly** [comment]
Note for anyone reviewing who looked at changes from the previous PR... I removed the master topology spread options.  Since this chart is single master only, spread constraints for the master wouldn't do anything.

**Comment by andresbono** [comment]
Thanks for submitting this again from your personal account.

File: bitnami/redis/templates/_helpers.tpl [hunk 1]
```diff
@@ -353,3 +353,26 @@ but Helm 2.9 and 2.10 does not support it, so we need to implement this if-else
...
+{{- if $message -}}
+{{-   printf "\nVALUES VALIDATION:\n%s" $message | fail -}}
+{{- end -}}
+{{- end -}}
+
+{{/* Validate values of Redis - spreadConstrainsts K8s version */}}
+{{- define "redis.validateValues.spreadConstraints" -}}
+{{- if and (semverCompare "<1.18-0" .Capabilities.KubeVersion.GitVersion) redis.validateValues.spreadConstraints -}}
+redis: spreadConstraints
+    Pod Topology Spread Constraints are only available on K8s  >= 1.8
```
Comment (andresbono) [inline bitnami/redis/templates/_helpers.tpl:375]: I have detected a minor typo here when testing it: 
```suggestion
{{- if and (semverCompare "<1.18-0" .Capabilities.KubeVersion.GitVersion) .Values.slave.spreadConstraints -}}
redis: spreadConstraints
    Pod Topology Spread Constraints are only available on K8s >= 1.8
```

Comment (jonstelly) [reply bitnami/redis/templates/_helpers.tpl:375]: And now that I look at that section it does a version check for `<1.18-0` but the feature seems to have been [added in 1.16](https://v1-16.docs.kubernetes.io/docs/concepts/workloads/pods/pod-topology-spread-constraints/).  Then the message says 1.8.  Should this also be changed to:

`semverCompare "<1.16-0"`
and
`only available on K8s >= 1.16`

Comment (andresbono) [reply bitnami/redis/templates/_helpers.tpl:375]: Thanks for pointing out. Yes, please, change that as well.

Comment (jonstelly) [reply bitnami/redis/templates/_helpers.tpl]: done and done.

**Comment by andresbono** [comment]
Thanks again for your contribution!

**Pull request merged**